ocaml/xapi/message_forwarding.ml

```diff
@@ -2630,7 +2630,13 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
             certificate_chain
         );
       in
-      tolerate_connection_loss fn success 30.
+      tolerate_connection_loss fn success 30.;
+      try
+        let _, _ = Forkhelpers.execute_command_get_output "/opt/xensource/libexec/alert-certificate-check" [] in
```

Comment:
The path should be defined in xapi_globs.ml.

Reply:
It's not, I'll define it, maybe we can slowly start using it as the path appears peppered around the code.

Reply:
I didn't mean `/opt/xensource/libexec`, but define the name of the command `alert-certificate-check` itself. In xapi_globs.ml, define it as `let alert_certificate_check = ref "alert-certificate-check"` and add it to `Resources.essential_executables` at the bottom of the file. This is used by logic in `Xcp_services.configure` (from xcp-idl), which is called at start up. This will search for the executables in `$PATH` and paths defined in the config file: we have `search-path = /opt/xensource/libexec:/opt/xensource/bin` in xapi.conf. See this being logged when xapi starts. So there is no need to have `/opt/xensource/libexec` in the code at all.